File: JobJarApi/src/main/java/org/jobjar/jobjarapi/domain/Offer.java
package org.jobjar.jobjarapi.domain;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.jobjar.jobjarapi.domain.responses.JustJoinItResponse;

import java.time.Instant;
import java.util.List;

@Entity
@Data
@NoArgsConstructor
@Table(name = "Offers")
public class Offer {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "offer_seq")
    private Long offerId;

    private String guid;

    private String title;

    private String slug;

    private List<Skill> requiredSkills;

    private WorkplaceType workplaceType;

    private ExperienceLevel experienceLevel;

    private Float minimalWage;

    private Float maximalWage;

    private Instant publishedAt;

    private Instant expiredAt;

    @JoinColumn(name = "company_id")
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    private Company company;

    public Offer(String guid,
                 String title,
                 String slug,
                 Float minimalWage,
                 Float maximalWage,
                 Instant publishedAt,
                 Instant expiredAt) {
        this.guid = guid;
        this.title = title;
        this.slug = slug;
        this.minimalWage = minimalWage;
        this.maximalWage = maximalWage;
        this.publishedAt = publishedAt;
        this.expiredAt = expiredAt;
    }
}

File: muncher_go/models/offer.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Offer struct {
	offerId         uuid.UUID
	externalId      uuid.UUID
	title           string
	slug            string
	requiredSkills  []Skill
	workplaceType   string
	experienceLevel string
	jobSite         string
	offerStatus     string
	minimalWage     float32
	maximalWage     float32
	publishedAt     time.Time
	expiredAt       time.Time
	company         *Company
}
